services/wallet/collectibles/manager.go

```diff
@@ -1027,3 +1028,80 @@ func (o *Manager) SearchCollections(ctx context.Context, chainID walletCommon.Ch
 	}
 	return nil, ErrNoProvidersAvailableForChainID
 }
+
+func (o *Manager) FetchCollectionSocialsAsync(contractID thirdparty.ContractID) error {
+	go func() {
+		defer o.checkConnectionStatus(contractID.ChainID)
+
+		provider, socials, err := o.fetchSocialsForCollection(context.Background(), contractID)
+		if err != nil || socials == nil {
+			log.Debug("FetchCollectionSocialsAsync failed for", "chainID", contractID.ChainID, "address", contractID.Address, "err", err)
+			return
+		}
+
+		socialsMessage := CollectionSocialsMessage{
+			ID:      contractID,
+			Socials: socials,
+		}
+
+		err = o.collectionsDataDB.SetCollectionSocialsData(provider, contractID, socials)
+		if err != nil {
+			log.Error("Error saving socials to DB: %v", err)
+			return
+		}
+
+		payload, err := json.Marshal(socialsMessage)
+		if err != nil {
+			log.Error("Error marshaling response: %v", err)
+			return
+		}
+
+		event := walletevent.Event{
+			Type:    EventGetCollectionSocialsDone,
+			Message: string(payload),
+		}
+
+		o.feed.Send(event)
+	}()
+
+	return nil
+}
+
+func (o *Manager) fetchSocialsForCollection(ctx context.Context, contractID thirdparty.ContractID) (string, *thirdparty.CollectionSocials, error) {
+	hasValidSocials, err := o.collectionsDataDB.HasValidSocialsForID(contractID)
+	if err != nil {
+		log.Debug("FetchCollectionSocialsAsync failed for", "chainID", contractID.ChainID, "address", contractID.Address, "err", err)
+		return "", nil, err
+	}
+
+	if !hasValidSocials {
+		cmd := circuitbreaker.Command{}
+		for _, provider := range o.providers.CollectibleDataProviders {
+			if !provider.IsChainSupported(contractID.ChainID) {
+				continue
+			}
+
+			provider := provider
+			cmd.Add(circuitbreaker.NewFunctor(func() ([]interface{}, error) {
+				socials, err := provider.FetchCollectionSocials(ctx, contractID)
+				if err != nil {
+					log.Error("FetchCollectionSocials failed for", "provider", provider.ID(), "chainID", contractID.ChainID, "err", err)
+				}
+				return []interface{}{provider.ID(), socials}, err
+			}))
+		}
+
+		if cmd.IsEmpty() {
+			return "", nil, ErrNoProvidersAvailableForChainID // lets not stop the group if no providers are available for the chain
+		}
+
+		cmdRes := o.getCircuitBreaker(contractID.ChainID).Execute(cmd)
+		if cmdRes.Error() != nil {
+			log.Error("fetchSocialsForCollection failed for", "chainID", contractID.ChainID, "err", cmdRes.Error())
+			return "", nil, cmdRes.Error()
+		}
+		return cmdRes.Result()[0].(string), cmdRes.Result()[1].(*thirdparty.CollectionSocials), cmdRes.Error()
+	}
+
+	return "", nil, errors.New("socials already fetched nothing to be done here")
```

Comment:
hmm I don't like the usage of an error here... Returning a non-nil error normally means there was some problem and whoever called this should handle it somehow (normally stopping whatever was being done). In this case it just means the caller doesn't need to fetch anything because we've got the needed data in the db already.

Moreover, for all other endpoints we do the fetching and the DB update (if successful) together (that way we ensure that whoever triggers a fetch will update the cache). 

So, IMO a tidier way to do this, with more separation of responsibilities, would be:
- `fetchSocialsForCollection` always fetches from some provider and updates the DB if successful
- other function `getOrFetchSocialsForCollection` returns what's on the DB if the entry is present, or returns `fetchSocialsForCollection` otherwise.
- `FetchCollectionSocialsAsync` calls `getOrFetchSocialsForCollection` and emits the event if successful

Reply:
Hey good suggestion! Have updated it this way :)